### DISCUSSION TITLE
Local copy of "select.js", and an option to toggle checking updates in admin settings

### DISCUSSION THREAD

**Discussion by dudeitssm** [Difegue/LANraragi#409] [Q&A] [answered]
Hi Difegue.

Thank you kindly for this software. I have it running perfectly inside a Raspberry Pi 3 with podman (docker, but better, imo).

I noticed that every time LRR is loaded, it makes two external (internet) requests: one to api.github.com, and one to cdn.datatables.net. This brings me to two minor feature requests:

Regarding the api.github.com request to check for updates, would it be possible to add a button in the admin config panel that would allow toggling this feature? I see it is part of a function listed [here](https://github.com/Difegue/LANraragi/blob/6a413a7fdd2a3e172d083467ee9be784cc21c5ec/public/js/index.js#L155). Something like the following, for example, would be a nice feature:
![example](https://user-images.githubusercontent.com/36095297/108301465-f5a37f80-7199-11eb-9075-2366bcc7a49b.png)

Regarding the select.js file, could it instead be locally stored, similar to how other js files are stored? (ie. inside public/js/vendor). Example, [within the index.html.tt2 file]( https://github.com/Difegue/LANraragi/blob/659ced5660d21718619280cd2053b552519ee5c7/templates/index.html.tt2#L29 ).

What are your opinions on these proposed changes? I could possibly take care of both requests, but may need your help with the first one--the second should be pretty straight forward for me.

**Comment by Difegue** [comment, marked as answer]
Thanks for bringing the select.js thing to my attention! 😳 
I believe it was inserted when I was iterating on the Jump to Page combobox, but the final code doesn't actually use it. I'll be pushing a commit to simply remove this include in index.html.tt2 later.  

For the update check, there's already a conditional to disable it [here](https://github.com/Difegue/LANraragi/blob/fdaae09ed8f94783e98afaad19eecd8bf905220a/templates/index.html.tt2#L157), but it currently checks for debug mode.  
I wouldn't mind having this moved to a dedicated pref at all.  

Adding new options is fairly straightforward! This recent commit https://github.com/Difegue/LANraragi/commit/1abce98c90f5e4eb588762f26537d2429422f350 deletes an option, but it gives you an idea of what you need. Here's a super quick breakdown:  

- Modify `lib/LANraragi/Model/Config.pm` to add a sub for your pref: Something like `sub get_autoupdate    { return &get_redis_conf( "autoupdate",      "1" ) }` will create a pref called "autoupdate" in the database defaulting to true.  
- Modify `lib/LANraragi/Controller/Config.pm` to expose this new preference to the template and handle its modification by the user
- Modify the proper html.tt2 template in ` templates/templates_config` to expose the pref in the proper section so the user can check/uncheck it (basically what you mocked up already)
- Expose `LANraragi::Model::Config->get_autoupdate` in the `render` function of `lib/LANraragi/Controller/Index.pm` so you can use it in `index.html.tt2` to disable the autoupdate 
- Done! 🎉

**Reply by dudeitssm** [reply]
Wow, you work fast :open_mouth: I saw the commit for the select.js was made a few hours ago.

And thank you for the steps :smile_cat: That should be all I need to test it out, and I'll mark the answer as solved.

**Reply by dudeitssm** [reply]
@Difegue am I on the right track? These are the changes I've added, across 4 files: https://github.com/Difegue/LANraragi/compare/dev...dudeitssm:dev

I'll try this out tomorrow morning on a VM to see if it works as expected: https://sugoi.gitbook.io/lanraragi/installing-lanraragi/source

Woops, just noticed I've not added the setting to toggle the update! Right now, it might just show the toggle button, but toggling it will not do anything. Derp... :man_facepalming:

**Reply by Difegue** [reply]
Looks good so far! Make sure to change the name/id of the select element in the templates from `devmode` to your new option, `autoupdate`. 

Past that, you're just missing the replacement in index.html.tt2, and handling the saving of the preference in Controller/Config.pm.

If your clone has access to GitHub codespaces, it might be easier to try your changes that way than through a VM! 😁
